### ISSUE TITLE
Broken links in glide_list when defined as Choice

### ISSUE THREAD

**Issue by m4te** [arnoudkooi/ServiceNow-Utils#76]
When a field is of List type, but does not have a Reference value - instead uses Choice definition + Choice Table, the generated link in the UI points to /null.do?sys_id=<choice label>. This naturally leads to a 404 HTTP error.

The link could either point to the sys_choice table, or alternatively it should not be a link.

**Comment by wiiaboo** [comment]
I'd agree with not being a link, since not all glide_lists can also have choices.

**Comment by arnoudkooi** [comment]
I thought I fixed that with that a while ago...
 https://github.com/arnoudkooi/ServiceNow-Utils/blob/master/inject.js#L949 

Anyway, will look at it and include it in the next update

**Comment by wiiaboo** [comment]
Even if the glide_list has a Reference Table, it may contain things other than sys_ids.

**Comment by m4te** [comment]
> Even if the glide_list has a Reference Table, it may contain things other than sys_ids.

I can confirm this, we experimented with setting Reference Key on List fields and it works, even though not very well. Some functions (i.e. setting value by direct GlideRecord attribute access) still set sys_id as value incorrectly.

The fix you are mentioning Arnoud looks like it should work, however ServiceNow threw us a curveball here, storing the table name in this case as "null" with type string. I've opened a pull request with the tiny fix.

**Comment by wiiaboo** [comment]
@arnoudkooi  On L955 below #77's changes, you could maybe use this instead, instead of hardcoding/assuming non-SysIDs are going to be emails or text with `@`:

```javascript
let hasReferenceTable = !table || table === 'null';
let sysIDregex = /[0-9a-f]{32}/i;
if (!hasReferenceTable || !sysIDRegex.test(values[i]))
    links.push(values[i]);
else
    links.push(`<a href="/${table}.do?sys_id=${values[i]}" target="_blank" />${labels[i]}</a>`);
```

**Comment by arnoudkooi** [comment]
PR?

**Comment by wiiaboo** [comment]
Sure, I can do it.

**Comment by arnoudkooi** [comment]
published in 4.1.1
